Standards/scs-03XX-v1-standard-roles.md

```diff
@@ -0,0 +1,271 @@
+
+The "creator" role will be kept for compatibility reasons concerning service integration.
+For example, the block storage service Cinder usually has a technical user in Keystone possessing the "creator" role in the "service" project.
+Moving such service accounts to the "member" role could introduce undesired access patterns in other APIs that otherwise don't accept the "creator" role but offer a lot of functionality to the "member" role by default.
+
+### Classification of the "manager" role
+
+The current RBAC rework in upstream OpenStack[^2] describes a "project-manager" persona utilizing a "manager" role on project scope to perform more privileged operations than "member" (see "Phase 3" of the document).
+This role is intended to be used across various OpenStack services.
+As of the OpenStack release 2024.1 this role is not implemented in any of the core services yet, only in Ironic with `enforce_new_defaults` enabled[^4].
+
+On the other hand, the SCS project is making use of this role to implement a Domain Manager persona (see the [SCS Domain Manager standard under "Related Documents"](#scs-domain-manager-standard)).
+
+As a result, the "manager" role has no effect outside of the Keystone Identity API until phase 3 of the RBAC rework is implemented upstream and this standard is migrated to the proper use of `enforce_new_defaults`.
+As such, it will be handled as a service-specific role for Keystone in this standard, not as a core role.
+This is to ensure that the effective scope of the role in SCS clouds is clearly documented and communicated to users.
+It might be elevated to a core role in future iterations of the standard when it is implemented in other services.
+
+[^4]: [Implementation of the "manager" role in Ironic for the 2024.1 release](https://github.com/openstack/ironic/blob/stable/2024.1/ironic/common/policy.py#L76-L82)
+
+### Open questions
+
+#### Incorporating future upstream defaults into this standard
+
+Due to the ongoing RBAC rework in upstream OpenStack[^2], not all changes which are to be introduced by it will be included in the first iteration of this standard to avoid prematurely adopting role and policy definitions which might still change before being stabilized or have unresolved compatibility issues with certain services.
+
+This results in a need of keeping this standard in sync once the upstream rework finishes.
+It is currently unknown when the upstream rework will conclude exactly and how this standard will need to be adjusted as a result.
+
+This primarily concerns the new scoping and defaults in `oslo.policy`:
+
+```ini
+[oslo_policy]
+enforce_new_defaults = True
+enforce_scope = True
+```
+
+As of the time of writing this standard, those options currently default to `False`[^3] for all OpenStack services.
```

Comment:
I thought they default to True for at least Nova, Glance and Neutron: https://etherpad.opendev.org/p/rbac-goal-tracking ?

Even though I wonder how these services work together with services with the old settings...

Reply:
Thank you for bringing this up! I was not aware of this and only looked up the defaults in the oslo.policy library[^1]. Furthermore, on the Secure RBAC page they stated "we decided to postpone the scope implementation"[^2] so I didn't expect something like this.

Turns out, Nova **overrides the oslo.policy defaults with its own defaults for the library**[^3]. Likely, Glance and Neutron do the same.

I need to check that and maybe need to rewrite parts of the standard once again ...

[^1]: https://github.com/openstack/oslo.policy/blob/c7083ec2c591e0f921d3532135e235695df0778f/oslo_policy/opts.py#L38

[^2]: https://governance.openstack.org/tc/goals/selected/consistent-and-secure-rbac.html#the-issues-we-are-facing-with-scope-concept

[^3]: https://github.com/openstack/nova/blob/3a53d715cd505258c1680c442b64c5bf7e182352/nova/policy.py#L44-L52

Reply:
Standard is now updated to reflect the actual state.